cpp/include/cudf/join.hpp

```diff
@@ -76,6 +76,8 @@ namespace cudf {
  * from `left_on` columns. Else, for every column in `left_on` and `right_on`,
  * an output column will be produced.  For each of these pairs (L, R), L
  * should exist in `left_on` and R should exist in `right_on`.
+ * @param[in] compare_nulls_equal is a bool that controls whether null join-key values
+ * should match (as per Pandas semantics), or not (as per SQL semantics)
```

Comment:
C++ documentation doesn't need to refer to other language semantics.

Reply:
Changed comments to leave out mention of SQL Join semantics.